libyui-rest-api/src/YMenuWidgetActionHandler.cc

```diff
@@ -0,0 +1,53 @@
+/*
+  Copyright (C) 2021 SUSE LLC
+
+  This library is free software; you can redistribute it and/or modify
+  it under the terms of the GNU Lesser General Public License as
+  published by the Free Software Foundation; either version 2.1 of the
+  License, or (at your option) version 3.0 of the License. This library
+  is distributed in the hope that it will be useful, but WITHOUT ANY
+  WARRANTY; without even the implied warranty of MERCHANTABILITY or
+  FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public
+  License for more details. You should have received a copy of the GNU
+  Lesser General Public License along with this library; if not, write
+  to the Free Software Foundation, Inc., 51 Franklin Street, Fifth
+  Floor, Boston, MA 02110-1301 USA
+*/
+
+#include <boost/algorithm/string.hpp>
+
+#define YUILogComponent "rest-api"
+
+#include <yui/YMenuItem.h>
+#include "YMenuWidgetActionHandler.h"
+
+YMenuItem* 
+YMenuWidgetActionHandler::findItem( std::vector<std::string>::iterator path_begin,
+                                    std::vector<std::string>::iterator path_end,
+                                    YItemConstIterator                 begin,
+                                    YItemConstIterator                 end ) const
+{
+    for ( YItemConstIterator it = begin; it != end; ++it )
+    {
+        YMenuItem * item = dynamic_cast<YMenuItem *>(*it);
+
+        if ( !item )
+            return nullptr;
+
+        if ( boost::erase_all_copy(item->label(), "&") == *path_begin )
+        {
+            if ( std::next( path_begin ) == path_end )
+            {
+                if ( item->hasChildren() )
+                    continue;
```

Comment:
I'm not sure if we should continue in this case, as we might want still to find item, even if it has children. I do understand that for some widgets we won't be able to select those (at least right now).
The only argument against it, is that we behave differently to other widgets. So worth at least documenting it, as in general such logic makes sense.

Reply:
This is copied from the YMenuWidget::find_item() method, but I'e removed the comment that describes the solution. If we already noticed that it is confusing, I'll add the comment back.

Reply:
I believe documentation is a better place for it, as we won't look into the code when using this plugin.